### PR TITLE
Image caching

### PR DESCRIPTION
**Pull request by scholvat**
This is based off of misc-bugs and has some dependencies on the privatising data PR, but thought I would open the PR so you can take a look at it. I'll update it once the other PRs are finalised and will be merged into this branch.

The purpose of this change is to locally cache the images (profile pictures and post pictures) to minimize data usage and API calls. It saves the images in local storage, and are retrieved when needed.

Currently this is configured to cache 30 profile pictures and 15 post pictures, which are 2 fairly arbitrary numbers that I thought sounded reasonable.

### CODE REVIEW

File: helpers/imageCache.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+							resolve(pic);
+						}
+					});
+				})
+				.catch(err => {
+					reject(err);
+				});
+		});
+	},
+	clearCache: function(){
```
Comment (neilbrub) [inline helpers/imageCache.js]: Any particular reason why we want to clear profile cache but not post cache?

**Comment by scholvat** [comment]
Has been updated with new API, and is now ready for review/merging. @neilbrub @picklechips could you take a look?

File: helpers/imageCache.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+	getProfilePicture: function(userID){
+		return new Promise(function(resolve, reject) {
+			//Try to get the item from the cache
+			profilePicCache.getItem(userID, function(err, cachedPic) {
+				if(err){
+					reject(err);
+				}else if(!cachedPic || !cachedPic.image || date.differenceInHours(new Date(),cachedPic.timeFetched)>24){
+					console.log("Missed profile!")
+					//cache miss, or over 24 hours old, so go fetch a new copy
+					ApiClient.get(`/users/${userID}/profilePicture`, {}).then(pic => {
```
Comment (neilbrub) [inline helpers/imageCache.js]: authorized? if not {} is unneeded (`get(endpoint, options={}) { ... }`)

File: helpers/imageCache.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+	getPostPicture: function(postID){
+		return new Promise(function(resolve, reject) {
+			//Try to get the item from the cache
+			postPicCache.getItem(postID, function(err, cachedPic) {
+				if(err){
+					reject(err);
+				}else if(!cachedPic || !cachedPic.image || date.differenceInHours(new Date(),cachedPic.timeFetched)>24){
+					console.log("Missed!")
+					//cache miss, or over 24 hours old, so go fetch a new copy
+					ApiClient.get(`/posts/${postID}/image`, {}).then(pic => {
```
Comment (neilbrub) [inline helpers/imageCache.js]: as above

**Comment by neilbrub** [comment]
LGTM besides comments above

**Comment by scholvat** [comment]
Good catch @neilbrub , fixed now 👍

**Pull request merged**